aidial_sdk/header_propagator.py

```diff
@@ -115,3 +103,21 @@ def instrumented_build_request(wrapped, instance, args, kwargs):
         wrapt.wrap_function_wrapper(
             httpx.AsyncClient, "build_request", instrumented_build_request
         )
+
+    def _modify_headers(
+        self, url: Any, headers: MutableMapping[str, str]
```

Comment:
Maybe we can change url type to str and resolve type casting in the instrument functions related to the clients?

Reply:
Indeed, fixed